src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/MinimumOSAttribute.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices
+{
+
+    /// <summary>
+    /// Records the minimum platform that is required in order to the marked thing.
+    /// <list type="bullet">
+    /// <item>
+    /// <description>When applied to an assembly, it means the entire assembly cannot be called
+    /// into on earlier versions. It records the TargetPlatformMinVersion property.</description>
+    /// </item>
+    /// <item>
+    /// <description> When applied to an API, it means the API cannot be called from an earlier
+    /// version.</description>
+    /// </item>
+    /// </list>
+    ///
+    /// In either case, the caller can either mark itself with MinimumPlatformAttribute
+    /// or guard the call with a platform check.
+    ///
+    /// The attribute can be applied multiple times for different operating systems.
+    /// That means the API is supported on multiple operating systems.
+    ///
+    /// A given platform should only be specified once.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Class |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple = true, Inherited = false)]
+    public sealed class MinimumOSAttribute : PlatformAttribute
```

Comment:
@terrajobst Should this be named `MinimumOSPlatformAttribute`? Was that what we landed on in #33331? (It wasn't clear to me what final name we landed on).

Reply:
Yes, the name we agreed on was `MinimumOSPlatformAttribute` https://github.com/dotnet/runtime/issues/33331#issuecomment-650326500